pkg/event-manager/handlers_driver.go

```diff
@@ -0,0 +1,256 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package eventmanager
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	apiclient "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vmware/dispatch/pkg/entity-store"
+	"github.com/vmware/dispatch/pkg/event-manager/gen/models"
+	driverapi "github.com/vmware/dispatch/pkg/event-manager/gen/restapi/operations/drivers"
+	"github.com/vmware/dispatch/pkg/secret-store/gen/client/secret"
+	"github.com/vmware/dispatch/pkg/trace"
+	"github.com/vmware/dispatch/pkg/utils"
+)
+
+var builtInDrivers = map[string]map[string]bool{
+	"vcenter": map[string]bool{
+		"vcenterurl": true,
+	},
+}
+
+// make sure the input includes all required config values
+func validateEventDriver(driver *Driver) error {
+	template, ok := builtInDrivers[driver.Type]
+	if !ok {
+		// custom driver, no validation
+		return nil
+	}
+
+	apiKeyAuth := apiclient.APIKeyAuth("cookie", "header", "cookie")
+	secrets := make(map[string]string)
+	for _, name := range driver.Secrets {
+		resp, err := SecretStoreClient().Secret.GetSecret(&secret.GetSecretParams{
+			SecretName: name,
+			Context:    context.Background(),
+		}, apiKeyAuth)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get secret %s from secret store", name)
```

Comment:
This is where our error handling sucks.  This should be returned to the user as a 400 with a clear error message which is consistent with the swagger generated 400 responses.

Reply:
Yeah. Also each service will use different approach for this... Can we figure out a unified way of handling errors? Is it even possible? Is defining global error types helpful?